plc4j/drivers/modbus/src/test/java/org/apache/plc4x/java/modbus/ManualModbusTCPDriverTest.java

```diff
@@ -69,22 +92,22 @@ public ManualModbusTCPDriverTest(String connectionString) {
     }
 
     public static void main(String[] args) throws Exception {
-        ManualModbusTCPDriverTest test = new ManualModbusTCPDriverTest("modbus-tcp://192.168.23.30");
+        ManualModbusTCPDriverTest test = new ManualModbusTCPDriverTest("modbus-tcp://10.10.1.200:10502?default-payload-byte-order=LITTLE_ENDIAN_BYTE_SWAP");
         test.addTestCase("holding-register:1:BOOL", new PlcBOOL(true)); // 0001
         test.addTestCase("holding-register:2:BYTE", new PlcBYTE(42)); // 2A
         //test.addTestCase("holding-register:3:WORD", new PlcWORD(42424)); // A5B8
-        test.addTestCase("holding-register:4:DWORD", new PlcDWORD(4242442424L)); // FCDE 88B8
+        test.addTestCase("holding-register:4:DWORD", new PlcDWORD(424244242L)); // 1949 7412
 //        test.addTestCase("holding-register:6:LWORD", new PlcLWORD(4242442424242424242L)); // FCDE 88B8 FCDE 88B8
         test.addTestCase("holding-register:10:SINT", new PlcSINT(-42)); // D6
         test.addTestCase("holding-register:11:USINT", new PlcUSINT(42)); // 2A
         test.addTestCase("holding-register:12:INT", new PlcINT(-2424)); // F688
         test.addTestCase("holding-register:13:UINT", new PlcUINT(42424)); // A5B8
-        test.addTestCase("holding-register:14:DINT", new PlcDINT(-242442424)); // F18C 9F48
-        test.addTestCase("holding-register:16:UDINT", new PlcUDINT(4242442424L));// FCDE 88B8
+        test.addTestCase("holding-register:14:DINT", new PlcDINT(-242442424)); // 1949 7412
+        test.addTestCase("holding-register:16:UDINT", new PlcUDINT(424244242L));// FCDE 88B8
         test.addTestCase("holding-register:18:LINT", new PlcLINT(-4242442424242424242L));// C51F D117 B2FB B64E
         test.addTestCase("holding-register:22:ULINT", new PlcULINT(4242442424242424242L));// 3AE0 2EE8 4D04 49B2
         test.addTestCase("holding-register:26:REAL", new PlcREAL(3.141593F));// 4049 0FDC
-        test.addTestCase("holding-register:28:LREAL", new PlcLREAL(2.71828182846D)); // 4005 BF0A 8B14 5FCF
+        test.addTestCase("holding-register:28:LREAL", new PlcLREAL(2.71)); // 4005 BF0A 8B14 5FCF
```

Comment:
Why cut off the fractional digits? if you enter the hex value here: https://baseconvert.com/ieee-754-floating-point you get the number that I had in there (I think this was the precision that Java has). So if your driver returns only 2.71, I would assume that something's wrong.

Reply:
Here again it is because the ModbusTools tool and an OPC-DA client against the DirectLink driver apply scientific notation, so it is easier to read 2.71

Reply:
I intentionally defined the test data to test the range of the values ... so do I understand it correctly, that in the tool you can only define two decimal digits?